Mathlib/FieldTheory/Separable.lean

```diff
@@ -554,72 +554,136 @@ theorem Algebra.isSeparable_iff :
+  ⟨fun _ ↦ AlgEquiv.Algebra.isSeparable e, fun _ ↦ AlgEquiv.Algebra.isSeparable e.symm⟩
 
+variable (F K) in
 instance Algebra.IsSeparable.isAlgebraic [Nontrivial F] [Algebra.IsSeparable F K] :
     Algebra.IsAlgebraic F K :=
   ⟨fun x ↦ (Algebra.IsSeparable.isIntegral F x).isAlgebraic⟩
 
+end
+
+section
+
+variable [Field L] [CommRing E] [Algebra F L]
+    [Algebra F E] [Algebra L E] [IsScalarTower F L E]
+
+/-- If `E / L / F` is an extension tower, `x : E` is separable over `F`, then it's also separable
+over `L`. -/
+theorem IsSeparable.tower_top
+    {x : E} (h : IsSeparable F x) : IsSeparable L x :=
+  h.map.of_dvd (minpoly.dvd_map_of_isScalarTower _ _ _)
+
+variable (F E) in
+theorem Algebra.isSeparable_tower_top_of_isSeparable [Algebra.IsSeparable F E] :
+    Algebra.IsSeparable L E :=
+  ⟨fun x ↦ IsSeparable.tower_top _ (Algebra.IsSeparable.isSeparable F x)⟩
+
+@[deprecated (since := "2024-08-06")]
+alias IsSeparable.of_isScalarTower := Algebra.isSeparable_tower_top_of_isSeparable
+
+end
+
 end CommRing
 
-instance Algebra.isSeparable_self (F : Type*) [Field F] : Algebra.IsSeparable F F :=
-  ⟨fun x => by
-    rw [IsSeparable, minpoly.eq_X_sub_C']
-    exact separable_X_sub_C⟩
+section Field
+
+variable (F : Type*) [Field F] {K E E' : Type*}
+
+section
+
+variable [Ring K] [Algebra F K]
+
+variable {F} in
+theorem isSeparable_algebraMap (x : F) : IsSeparable F (algebraMap F K x) :=
+  Polynomial.Separable.of_dvd (Polynomial.separable_X_sub_C (x := x))
+    (minpoly.dvd F (algebraMap F K x) (by simp only [map_sub, aeval_X, aeval_C, sub_self]))
+
+instance Algebra.isSeparable_self : Algebra.IsSeparable F F :=
+  ⟨fun _ => isSeparable_algebraMap _⟩
+
+variable [IsDomain K] [FiniteDimensional F K] [CharZero F]
+
+theorem IsSeparable.of_finite (x : K) : IsSeparable F x :=
+  (minpoly.irreducible <| .of_finite F x).separable
 
+variable (K) in
 -- See note [lower instance priority]
 /-- A finite field extension in characteristic 0 is separable. -/
-instance (priority := 100) Algebra.IsSeparable.of_finite (F K : Type*) [Field F] [Field K]
-    [Algebra F K] [FiniteDimensional F K] [CharZero F] : Algebra.IsSeparable F K :=
-  ⟨fun x => (minpoly.irreducible <| .of_finite F x).separable⟩
+protected instance (priority := 100) Algebra.IsSeparable.of_finite : Algebra.IsSeparable F K :=
+  ⟨fun _ => _root_.IsSeparable.of_finite _ _⟩
 
-section IsSeparableTower
+end
 
-/-- If `R / K / A` is an extension tower, `x : R` is separable over `A`, then it's also separable
-over `K`. -/
-theorem IsSeparable.of_isScalarTower {A : Type*} [CommRing A]
-    (K : Type*) [Field K] [Algebra A K] {R : Type*} [CommRing R] [Algebra A R] [Algebra K R]
-    [IsScalarTower A K R] {x : R} (h : IsSeparable A x) : IsSeparable K x :=
-  h.map.of_dvd (minpoly.dvd_map_of_isScalarTower _ _ _)
-
-variable (F K E : Type*) [Field F] [Field K] [Field E] [Algebra F K] [Algebra F E] [Algebra K E]
-  [IsScalarTower F K E]
+section
```

Comment:
Why a new `section` here? Why don't you need a section name? Perhaps `ScalarTower`?

Reply:
Here in this part of the file, every single theorem has a different list of variables (E F being Ring or Field, ...). I open new sections to control these variables.

Reply:
Here in this part of the file, every single theorem has a different list of variables (E F being Ring or Field, ...). I open new sections to control these variables.

Reply:
I've added names for these sections. Thanks!